lib/instrumentation/modules/@elastic/elasticsearch.js

```diff
@@ -0,0 +1,124 @@
+'use strict'
+
+// Instrument the @elastic/elasticsearch module.
+//
+// This uses to 'request' and 'response' events from the Client (documented at
+// https://www.elastic.co/guide/en/elasticsearch/client/javascript-api/current/observability.html)
+// to hook into all ES server interactions.
+
+const { getDBDestination } = require('../../context')
+
+// TODO: ask Tomas about apiName addition to RequestEvent
+const pathIsAQuery = /\/_((search|msearch)(\/template)?|count)$/
```

Comment:
Calling out some offline discussions here -- this test means that we'll only capture the full request bodies for REST queries whose paths match the following _exactly_. 

```
_search
_msearch
_search/template
_msearch/template
_count
```

We may want to expand this after discussions with stakeholders.

Reply:
I've also asked @delvedor about perhaps changing elasticsearch-js to provide a field in the 'request' event that is better than just regex matching the path (which can have user data in it).

I've also got this change coming:

```
// Allow the following URL path components:
// - _search
// - _msearch
// - _search/template
// - _msearch/template
// - _count
// - _async_search
const pathIsAQuery = /\/(_search|_msearch|_count|_async_search)(\/|$)/
```

to add `_async_search` because Kibana is using that.